src/Http/Http/test/Features/FormFeatureTests.cs

```diff
@@ -82,6 +83,49 @@ public async Task ReadFormAsync_SimpleData_ReturnsParsedFormCollection(bool buff
             await responseFeature.CompleteAsync();
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task ReadFormAsync_SimpleData_ReplacePipeReader_ReturnsParsedFormCollection(bool bufferRequest)
+        {
+            var formContent = Encoding.UTF8.GetBytes("foo=bar&baz=2");
+            var context = new DefaultHttpContext();
+            var responseFeature = new FakeResponseFeature();
+            context.Features.Set<IHttpResponseFeature>(responseFeature);
+            context.Request.ContentType = "application/x-www-form-urlencoded; charset=utf-8";
+
+            var pipe = new Pipe();
+            await pipe.Writer.WriteAsync(formContent);
+            pipe.Writer.Complete();
+
+            context.Request.BodyPipe = pipe.Reader;
+
+            IFormFeature formFeature = new FormFeature(context.Request, new FormOptions() { BufferBody = bufferRequest });
+            context.Features.Set<IFormFeature>(formFeature);
+
+            var formCollection = await context.Request.ReadFormAsync();
+
+            Assert.Equal("bar", formCollection["foo"]);
+            Assert.Equal("2", formCollection["baz"]);
+
+            // TODO this returns a null stream because we don't wrap stream/pipereader like we do in Kestrel
```

Comment:
So this is a test issue. Remove the code completely then?

Reply:
Once we fix how wrapping works in HttpAbstractions, I want to revert this code.